hpx/apply.hpp

```diff
@@ -33,6 +36,10 @@ namespace hpx { namespace detail
     // Define apply() overloads for plain local functions and function objects.
     // dispatching trait for hpx::apply
 
+#if !defined(HPX_HAVE_DISTRIBUTED_RUNTIME)
+    template <typename Func, typename Enable = void>
+    struct apply_dispatch;
+#endif
```

Comment:
Where is this template declared if `HPX_HAVE_DISTRIBUTED_RUNTIME` is actually defined? Should we really separate that?

Reply:
No, it doesn't make sense. Especially after #4478 the base template will be in `async_base` and always declared. I'll change this once that's in master and this is rebased.